checkpoint/recovery_test.go

```diff
@@ -255,7 +256,7 @@ func validateAndPreserveData(
 	)
 	mfetch.EXPECT().GetAtxs(gomock.Any(), gomock.Any(), gomock.Any()).AnyTimes()
 	for i, vatx := range deps {
-		encoded, err := codec.Encode(vatx)
+		encoded, err := codec.Encode(wire.ActivationTxToWireV1(vatx.ActivationTx))
```

Comment:
Recovery will not use the domain type but the wire type in my opinion. Meaning that if a checkpoint is created it will only contain the ATX blobs which recovery decodes to the wire type (and then restores by passing it by the activation handler).

Recovery only needs to make sure that the ATXs are persisted in publication order so that the handler can verify them correctly during restore.

Reply:
This also came up in a discussion on a different PR here: https://github.com/spacemeshos/go-spacemesh/pull/5791#discussion_r1561281266

In one of the next PRs we need to update checkpoint to store the ATXs just as bytes and not wire types.